cli/packages/cmd/secrets.go

```diff
@@ -251,7 +266,16 @@ var secretsSetCmd = &cobra.Command{
 				util.HandleError(err, "unable to encrypt your secrets")
 			}
 
-			if existingSecret, ok := secretByKey[key]; ok {
+			var existingSecret models.SingleEnvironmentVariable
+			var doesSecretExist bool
+
+			if secretType == util.SECRET_TYPE_SHARED {
+				existingSecret, doesSecretExist = sharedSecretMapByName[key]
+			} else {
+				existingSecret, doesSecretExist = personalSecretMapByName[key]
+			}
```

Comment:
I would probably do an `else if secret.Type == util.SECRET_TYPE_SHARED`. And then and `else` statement that throws an error if the secretType is not `shared` or `personal`, because it's possible that the user has unknowingly made a typo when they're typing the command.